internal/handlers/common/update.go

```diff
@@ -918,23 +918,47 @@ func validateOperatorKeys(command string, docs ...*types.Document) error {
 				)
 			}
 
-			if _, ok := seen[path.Prefix()]; ok {
-				return newUpdateError(
-					commonerrors.ErrConflictingUpdateOperators,
-					fmt.Sprintf(
-						"Updating the path '%[1]s' would create a conflict at '%[1]s'", key,
-					),
-					command,
-				)
+			for _, oldPath := range visistedPaths {
+				if checkSlicePrefix(oldPath.Slice(), nextPath.Slice()) {
+					return newUpdateError(
+						commonerrors.ErrConflictingUpdateOperators,
+						fmt.Sprintf(
+							"Updating the path '%[1]s' would create a conflict at '%[1]s'", key,
+						),
+						command,
+					)
+				}
 			}
-
-			seen[path.Prefix()] = struct{}{}
+			visistedPaths = append(visistedPaths, &nextPath)
 		}
 	}
 
 	return nil
 }
 
+// checkSlicePrefix returns true if one slice is the beginning of another slice.
+// The example of slice prefix: arr1 = ["a","b","c"] arr2 = ["a","b"];
+// If both slices are empty, it returns true. If one slice is empty and another slice is not, it returns false.
+func checkSlicePrefix(arr1, arr2 []string) bool {
+	target, prefix := arr1, arr2
+
+	if len(target) < len(prefix) {
+		target, prefix = prefix, target
+	}
+
+	if len(prefix) == 0 {
+		return len(target) == 0
```

Comment:
Codecov thinks this is not tested, could this be tested?

Reply:
Hmm not really (with integration tests), as path cannot be empty (empty string would result in slice with 1 item).

Reply:
Okie, I though it might work with `.` path. But might not.